### ISSUE TITLE
[RFE] include pidstat data in sa data files

### ISSUE THREAD

**Issue by pschiffe** [sysstat/sysstat#111]
It would be useful to have historical pidstat data, to investigate which process misbehaved last night.

**Comment by sysstat** [comment]
Yes I can understand that having historical pidstat data can help troubleshoot problems.
But hey, I consider the idea to include those data in sa data files to be frightening...! Please tell me, you were just kidding?

**Comment by pschiffe** [comment]
Yeah, I thought you wouldn't like it from the techn. point of view. :-) I put it here because I think it's a really good feature to have. Maybe if it would be optional, off by default.. I don't know.. I don't want to scare you, so feel free to close this if you don't like it. (but I'm already thinking about optimizations like store the process name with params only once in data file, and then, it's again only just numbers.. ^^)

**Comment by AkhilJalagam** [comment]
presently, I am using `telegraf -> influxdb -> grafana` for historical metrics data. but still `pidstat` is simple compared to grafana system. it's good to have sysstat/pidstat historical metrics.

**Comment by MrPippin66** [comment]
Including "pidstat" data in "sar" data is a data management problem. But I have to cast my vote to have the option of gathering this data for 'sar' reporting as well. It would really help to have the option after the fact to see what processes were consuming CPU resources during a given interval.

**Comment by leahneukirchen** [comment]
How about pidstat is trucated to the top 5/10/20 processes? Then the data set should be managable.

**Comment by natoscott** [comment]
There is a bit of a roundabout solution using PCP for anyone needing this functionality:

- record sa data as usual
- run a pmlogger(1) process recording just process metrics
- at the end of each day, use sadf -O to create a PCP archive containing the sa data
- merge the two archives using pmlogextract(1) to create <combined-daily-archive>
- use pcp-pidstat --archive <combined-daily-archive>

A few years ago we started a project (myself, @sysstat @stevekay) to have sadc optionally write PCP archives directly (we still have the notes somewhere I'm sure, if anyone is interested in restarting this effort).

If we had that capability, it'd be relatively easy to add per-process metrics recording into sar (and many other metrics of course - GPUs, Prometheus/OpenMetrics/OpenTelemetry, custom application metrics, etc).
